pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstance.java

```diff
@@ -67,6 +69,13 @@ public JavaInstance(InstanceConfig config, Object userClassObject,
     }
 
     public JavaExecutionResult handleMessage(MessageId messageId, String topicName, Object input) {
+        if (source instanceof PulsarSource) {
```

Comment:
This is actually instantiating a context object per message. I think that should be avoided.